### PR TITLE
[CleanupUnneededNullsafeOperatorRector] Wip adding new rule fix #7781

### PR DESCRIPTION
**Pull request by JohJohan**

### CODE REVIEW

File: rules/Php80/Rector/NullsafeMethodCall/CleanupUnneededNullsafeOperatorRector.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    /**
+     * @param NullsafeMethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (!$node->var instanceof MethodCall) {
+            return null;
+        }
+
+        $returnNode = $this->returnStrictTypeAnalyzer->resolveMethodCallReturnNode($node->var);
```
Comment (ghost) [inline rules/Php80/Rector/NullsafeMethodCall/CleanupUnneededNullsafeOperatorRector.php]: This now returns `null` for all my fixtures am i doing something wrong?

File: rules/Php80/Rector/NullsafeMethodCall/CleanupUnneededNullsafeOperatorRector.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        if (null === $returnNode) {
+            return null;
+        }
+
+        // Remove not needed Nullsafe for method call.
+        $node = $node->var;
```
Comment (ghost) [inline rules/Php80/Rector/NullsafeMethodCall/CleanupUnneededNullsafeOperatorRector.php]: I now remove `->getReplace();` as well is there a way to keep that maybe i should get parent call and get it from there?

**Comment by ghost** [comment]
Fixes https://github.com/rectorphp/rector/issues/7781

File: rules/Php80/Rector/NullsafeMethodCall/CleanupUnneededNullsafeOperatorRector.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        return $node;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::NULLSAFE_OPERATOR;
```
Comment (ghost) [inline rules/Php80/Rector/NullsafeMethodCall/CleanupUnneededNullsafeOperatorRector.php]: Is this the way to make sure your rule only runs when we support null safe operator or should there be done something extra?

**Review by samsonasik — CHANGES_REQUESTED** [review]
This should go to `CodeQuality` sets

**Comment by ghost** [comment]
@samsonasik i moved to `CodeQuality`

**Comment by TomasVotruba** [comment]
This looks good :+1: Thank you for your patience, as I went trough dozens of other PRs.

Could you rebase and make the CI pass? I'll give it a priority now :slightly_smiling_face:

**Comment by TomasVotruba** [comment]
@samsonasik Could you rebase and handle this one as well? It's been opened a few months, so we can handle the changes for the @johanadivare

**Comment by samsonasik** [comment]
I will try 👍

**Comment by samsonasik** [comment]
It has too many conflict with rebase, merge main to this branch is the way to go

**Comment by samsonasik** [comment]
@JohJohan I will try continue your PR at https://github.com/rectorphp/rector-src/pull/3767

**Pull request closed without merging**